app/gui/src/controller/searcher/input.rs

```diff
@@ -371,16 +380,117 @@ impl InsertedSuggestion {
     }
 }
 
+
+// === Insert Context ===
+
+/// A helper abstraction responsible for generating proper code for the suggestion.
+///
+/// If the context contains a qualified name, we want to reuse it in the generated code sample.
+/// For example, if the user types `Foo.Bar.` and accepts a method `Foo.Bar.baz` we insert
+/// `Foo.Bar.baz` with `Foo` import instead of inserting `Bar.baz` with `Bar` import.
+struct InsertContext<'a> {
+    suggestion: &'a action::Suggestion,
+    context:    Option<ast::opr::Chain>,
+}
+
+impl InsertContext<'_> {
+    /// Whether the context of the edited expression contains a qualified name. Qualified name is a
+    /// infix chain of `ACCESS` operators with identifiers.
+    fn has_qualified_name(&self) -> bool {
+        if let Some(ref context) = self.context {
+            let every_operand_is_name = context
+                .enumerate_operands()
+                .flatten()
+                .all(|opr| matches!(opr.item.arg.shape(), ast::Shape::Cons(_)));
```

Comment:
Technically the Qualified Names could be also lowercase, what resolves to `ast::Shape::Var` variant (most common example is `local` namespace, but modules theoretically could be lowercase as well AFAIK).

Reply:
👍🏻 Fixed.